### PR TITLE
Code migration

### PR DESCRIPTION
**Pull request by gregsdennis**
This migrates the code (with some modifications) from our private repositories.

### CODE REVIEW

**Review by erwan-joly — APPROVED** [review]
LGTM, wonder if the namespace should stay Pushpay or be PushpayLabs

**Comment by gregsdennis** [comment]
The DynamoDbProvisioner is `Pushpay`, so I just followed that.

**Comment by gregsdennis** [comment]
Will set up CI in the next PR.

File: SemVerAnalyzer.sln [hunk 1]
```diff
@@ -0,0 +1,63 @@
+﻿
+Microsoft Visual Studio Solution File, Format Version 12.00
```
Comment (chris-findlay) [inline SemVerAnalyzer.sln:2]: I wonder if these files always start with a blank line by way of detecting which linebreaks are present...

File: source/SemVerAnalyzer/Assembly/MethodSignatureComparer.cs [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+			}
+
+			if (x == null || y == null) {
+				return false;
+			}
+
+			return EqualNames(x, y) &&
+				   EqualGenericParameters(x, y) &&
+				   EqualTypes(x.ReturnType, y.ReturnType) &&
+				   EqualParameters(x, y);
```
Comment (chris-findlay) [inline source/SemVerAnalyzer/Assembly/MethodSignatureComparer.cs:25]: looks like there's some code in here that doesn't match the checked in settings 😛

File: source/SemVerAnalyzer/Assembly/MethodSignatureComparer.cs [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+		}
+
+		public int GetHashCode(MethodDef obj)
+		{
+			return obj.Name.GetHashCode() ^ obj.Parameters.Count; // everything else would be too cumbersome
```
Comment (chris-findlay) [inline source/SemVerAnalyzer/Assembly/MethodSignatureComparer.cs:30]: See also `HashCode.Combine(obj.Name, obj.Parameters)`

File: source/SemVerAnalyzer/Engine/SemverExtensions.cs [hunk 4]
```diff
@@ -0,0 +1,52 @@
+using System;
+using System.Text.RegularExpressions;
+
+namespace Pushpay.SemVerAnalyzer.Engine
+{
+	public static class SemverExtensions
+	{
+		static readonly Regex _versionFormat = new Regex(@"^(?<major>[0-9]+)\.(?<minor>[0-9]+)\.(?<patch>[0-9]+)((?<prerelease>-\w*)|(?<trailer>\.0))?(?<githash>\+[0-9a-f]{40})?$");
```
Comment (chris-findlay) [inline source/SemVerAnalyzer/Engine/SemverExtensions.cs:8]: `\d`===`[0-9]`

File: source/SemVerAnalyzer/Nuget/NugetClient.cs [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+		public async Task<byte[]> GetAssemblyBytesFromPackage(string packageName, List<string> comments)
+		{
+
+			try {
+				using var client = new HttpClient();
+				using var request = new HttpRequestMessage(HttpMethod.Get, Path.Join(_config.RepositoryUrl, packageName));
+				using var response = await client.SendAsync(request);
+				if (!response.IsSuccessStatusCode) {
+					comments.Add("Error contacting ProGet.  If working outside of the office, please ensure you are on the VPN.");
```
Comment (chris-findlay) [inline source/SemVerAnalyzer/Nuget/NugetClient.cs:27]: Do we want to mention `ProGet` here or something more generic?

Comment (chris-findlay) [reply source/SemVerAnalyzer/Nuget/NugetClient.cs:27]: Also below.

**Review by chris-findlay — COMMENTED** [review]
Just a few questions.

**Comment by gregsdennis** [comment]
@chris-findlay, this was just a port from private code. I'm primarily concerned with secrets right now.  If there are other changes you suggest, please open an issue. This is now open source.

**Pull request merged**